qt/scientific_interfaces/Indirect/test/ISISEnergyTransferModelTest.h

```diff
@@ -0,0 +1,605 @@
+    auto reductionAlgorithm = makeReductionAlgorithm();
+
+    IETGroupingData groupingData(IETGroupingType::CUSTOM, 2, "map_file", "1,2,3");
+    IETConversionData conversionData(1.0, 1, 5);
+    model->setGroupingProperties(reductionAlgorithm, groupingData, conversionData);
+
+    TS_ASSERT_EQUALS(reductionAlgorithm->getPropertyValue("GroupingString"), "1,2,3");
+    TS_ASSERT_EQUALS(reductionAlgorithm->getPropertyValue("MapFile"), "");
+  }
+
+  void testSetGroupingPropertiesWithDefaultGrouping() {
+    auto model = makeModel();
+    auto reductionAlgorithm = makeReductionAlgorithm();
+
+    IETGroupingData groupingData(IETGroupingType::DEFAULT, 2, "map_file", "1,2,3");
+    IETConversionData conversionData(1.0, 1, 5);
+    model->setGroupingProperties(reductionAlgorithm, groupingData, conversionData);
+
+    TS_ASSERT_EQUALS(reductionAlgorithm->getPropertyValue("GroupingString"), "");
+    TS_ASSERT_EQUALS(reductionAlgorithm->getPropertyValue("MapFile"), "");
+  }
+
+  void testSetGroupingPropertiesWithGroupsGrouping() {
+    auto model = makeModel();
+    auto reductionAlgorithm = makeReductionAlgorithm();
+
+    IETGroupingData groupingData(IETGroupingType::GROUPS, 2, "map_file", "1,2,3");
+    IETConversionData conversionData(1.0, 1, 5);
+    model->setGroupingProperties(reductionAlgorithm, groupingData, conversionData);
+
+    TS_ASSERT_EQUALS(reductionAlgorithm->getPropertyValue("GroupingString"), "1-2,3-4,5-5");
+    TS_ASSERT_EQUALS(reductionAlgorithm->getPropertyValue("MapFile"), "");
+  }
+
+  void testGetOutputGroupName() {
+    auto model = makeModel();
+
+    InstrumentData instData("instrument", "analyser", "reflection");
+    std::string inputFiles("1234, 1235");
+    std::string outputName = model->getOuputGroupName(instData, inputFiles);
+
+    TS_ASSERT_EQUALS(outputName, "instrument1234, 1235_analyser_reflection_Reduced");
+  }
+
+  void testRunIETAlgorithm() {
+    MantidQt::API::BatchAlgorithmRunner *batch = new MantidQt::API::BatchAlgorithmRunner(nullptr);
+
+    IETInputData inputData("input_workspace1, input_workspace2", true, false, true, "calibration_workspace");
+    IETConversionData conversionData(1.0, 1, 2);
+    IETGroupingData groupingData(IETGroupingType::DEFAULT, 2, "map_file");
+    IETBackgroundData backgroundData(true, 0, 1);
+    IETAnalysisData analysisData(true, 2.5, true, 2.0);
+    IETRebinData rebinData(true, "Multiple", 0.0, 0.0, 0.0, "1,2");
+    IETOutputData outputData(false, false);
+
+    IETRunData runData(inputData, conversionData, groupingData, backgroundData, analysisData, rebinData, outputData);
+
+    InstrumentData instData("instrument", "analyser", "reflection");
+
+    m_model->runIETAlgorithm(batch, instData, runData);
+
+    // Wait for the algorithm to finish
+    std::this_thread::sleep_for(std::chrono::seconds(1));
+
+    TS_ASSERT_EQUALS(AnalysisDataService::Instance().doesExist("outputWS"), true);
+    if (AnalysisDataService::Instance().doesExist("outputWS")) {
```

Comment:
I dont think you need this as the check is effectively done above in the assert does exist

Reply:
Without this check, I was getting an error in `Mantid::API::AnalysisDataService::Instance().retrieveWS` because it couldn't find the workspace in the ADS. So that is why I've added this check

Reply:
thats very strange

Reply:
The error happens because if `doesExist` returns false that means `outputWs` is not in the ADS so this is why I added the check